odoo/addons/base/models/ir_http.py

```diff
@@ -118,12 +120,28 @@ class IrHttp(models.AbstractModel):
     _name = 'ir.http'
     _description = "HTTP Routing"
 
+    @classmethod
+    def _slug(cls, value: models.BaseModel | str) -> str:
+        try:
+            return str(value.id)
+        except AttributeError:
+            pass
+        slug_str = unicodedata.normalize('NFKD', str(value)).encode('ascii', 'ignore').decode('ascii')
+        return re.sub(r'\W+', '-', slug_str or '')
```

Comment:
it used to use `slugify` which itself was using `slugify_one` which itself was doing as follow:
```py
    if slugify_lib:
        # There are 2 different libraries only python-slugify is supported
        try:
            return slugify_lib.slugify(s, max_length=max_length)
        except TypeError:
            pass
    uni = unicodedata.normalize('NFKD', s).encode('ascii', 'ignore').decode('ascii')
    slug_str = re.sub(r'[\W_]', ' ', uni).strip().lower()
    slug_str = re.sub(r'[-\s]+', '-', slug_str)
    return slug_str
```
why don't we use the slugify_lib here? why are the regexp substitutions different?

Reply:
=> `re.sub(r'[\W_]+', '-', uni).strip('-').lower()`

Reply:
in the original regexpes, multiple following `-` would collapse as one, it is not the case anymore

Reply:
Reminder: we need to keep the support of negative id in the slug ...

Reply:
Let's not change this without unittests

Reply:
@Julien00859 
> in the original regexpes, multiple following `-` would collapse as one, it is not the case anymore

Their are no change. `-` is a `\W`

```
re.sub(r'[\W_]', ' ', uni).strip().lower()
re.sub(r'[-\s]+', '-', slug_str)
```
"  a--z+gg_54   -5 -"  => "a  z gg 54    5" => "a-z-gg-54-5"

is equal to:

 `re.sub(r'[\W_]+', '-', uni).strip('-').lower()`
"  a--z+gg_54   -5 -"  => "a-z-gg-54-5"

@JKE-be The negatif id don't go through this part of code. The id is systematically added afterwards. I will see if there is a test for negative IDs, if not I will add one.

Reply:
The tests are in: `TestTitleToSlug`
I added tests for the negative id slug